tartufo/config.py

```diff
@@ -195,22 +196,34 @@ def configure_regexes(
     return rules
 
 
-def load_rules_from_file(rules_file: TextIO) -> Dict[str, Pattern]:
+def load_rules_from_file(rules_file: TextIO) -> Dict[str, Rule]:
     """Load a set of JSON rules from a file and return them as compiled patterns.
 
     :param rules_file: An open file handle containing a JSON dictionary of regexes
     :raises ValueError: If the rules contain invalid JSON
     """
-    regexes = {}
+    rules: Dict[str, Rule] = {}
     try:
         new_rules = json.load(rules_file)
     except json.JSONDecodeError as exc:
         raise ValueError(
             "Error loading rules from file: {}".format(rules_file.name)
         ) from exc
-    for rule in new_rules:
-        regexes[rule] = re.compile(new_rules[rule])
-    return regexes
+    for rule_name in new_rules:
+        rule_definition = new_rules[rule_name]
```

Comment:
```suggestion
    for rule_name, rule_definition in new_rules.items():
```
This isn't an absolutely necessary change, but a nice short-hand and common pattern.

Reply:
This is another one of those "somehow this code is being reviewed at a higher level of diligence than the original code"... https://github.com/godaddy/tartufo/blob/master/tartufo/config.py#L212

Reply:
As we continue to onboard more contributors we should endeavor to continuously uplift the code quality, i.e. constantly raise the bar. The original code was done under a specific set of conditions. Since then we have been making a concerted effort to treat this less in that manner and more as the open source project that it is.

Reply:
I agree, however this PR was not intended to uplift code. This follows the original pattern of the code. I don't recommend you include nor mandate contributors to "uplift" existing code. This type of feedback only blocks added value to this tool.

Reply:
I'm not asking for a rewrite of untouched code. Since lines 211-213 are already being modified, why not change them to the new convention? Does it change the readability that much in comparison to the old pattern?

Reply:
I would suggest that any time a contribution touches a section of code, and that code can be improved on, that improvement should take place naturally.  The goal should be for everyone to incrementally improve the code through contributions where possible.  That being said, the expectation is not to rewrite the code every time a PR is submitted, but to simply be a good contribution citizen and help improve the overall quality of the code.